src/test/ui/enum/union-in-enum.rs

```diff
@@ -0,0 +1,5 @@
+enum A { union } //~ WARNING
```

Comment:
Could you add a comment here explaining what this is testing? Just something like:

> This test checks that the `union` keyword is accepted as the name of an enum variant when not followed by an identifier.

Reply:
I was really stucked in testcase part.
.It is my first contribution.
Thank you for all your suggestions.